cluster-autoscaler/expander/grpcplugin/grpc_client_test.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpcplugin
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/expander/grpcplugin/protos"
+	"k8s.io/autoscaler/cluster-autoscaler/expander/mocks"
+	. "k8s.io/autoscaler/cluster-autoscaler/utils/test"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider/test"
+	"k8s.io/autoscaler/cluster-autoscaler/expander"
+
+	_ "github.com/golang/mock/mockgen/model"
+)
+
+var (
+	nodes = []*v1.Node{
+		BuildTestNode("n1", 1000, 1000),
+		BuildTestNode("n2", 1000, 1000),
+		BuildTestNode("n3", 1000, 1000),
+		BuildTestNode("n4", 1000, 1000),
+	}
+
+	eoT2Micro = expander.Option{
+		Debug:     "t2.micro",
+		NodeGroup: test.NewTestNodeGroup("my-asg.t2.micro", 10, 1, 1, true, false, "t2.micro", nil, nil),
+	}
+	eoT2Large = expander.Option{
+		Debug:     "t2.large",
+		NodeGroup: test.NewTestNodeGroup("my-asg.t2.large", 10, 1, 1, true, false, "t2.large", nil, nil),
+	}
+	eoT3Large = expander.Option{
+		Debug:     "t3.large",
+		NodeGroup: test.NewTestNodeGroup("my-asg.t3.large", 10, 1, 1, true, false, "t3.large", nil, nil),
+	}
+	eoM44XLarge = expander.Option{
+		Debug:     "m4.4xlarge",
+		NodeGroup: test.NewTestNodeGroup("my-asg.m4.4xlarge", 10, 1, 1, true, false, "m4.4xlarge", nil, nil),
+	}
+	options = []expander.Option{eoT2Micro, eoT2Large, eoT3Large, eoM44XLarge}
+
+	grpcEoT2Micro = protos.Option{
+		NodeGroupId: eoT2Micro.NodeGroup.Id(),
+		NodeCount:   int32(eoT2Micro.NodeCount),
+		Debug:       eoT2Micro.Debug,
+		Pod:         eoT2Micro.Pods,
+	}
+	grpcEoT2Large = protos.Option{
+		NodeGroupId: eoT2Large.NodeGroup.Id(),
+		NodeCount:   int32(eoT2Large.NodeCount),
+		Debug:       eoT2Large.Debug,
+		Pod:         eoT2Large.Pods,
+	}
+	grpcEoT3Large = protos.Option{
+		NodeGroupId: eoT3Large.NodeGroup.Id(),
+		NodeCount:   int32(eoT3Large.NodeCount),
+		Debug:       eoT3Large.Debug,
+		Pod:         eoT3Large.Pods,
+	}
+	grpcEoM44XLarge = protos.Option{
+		NodeGroupId: eoM44XLarge.NodeGroup.Id(),
+		NodeCount:   int32(eoM44XLarge.NodeCount),
+		Debug:       eoM44XLarge.Debug,
+		Pod:         eoM44XLarge.Pods,
+	}
+)
+
+func TestPopulateOptionsForGrpc(t *testing.T) {
```

Comment:
Please consider a table-based test to cover various inputs, including edge cases (ex. empty list of options).

Reply:
added some tests, as requested